### ISSUE TITLE
How to constraint sin functions for a simple damped pendulum

### ISSUE THREAD

**Issue by TheUltimate101** [dynamicslab/pysindy#602]
Hi, i want to constraint to find the function using data of a damped pendulum. I am having trouble using linear constraint to put the sin theta term of the omega dot to be -9.81 and theta dot to be fixed to be omega. I will be very thankful for any help i can get.

I am attaching my trial code below, i am new to pysindy

`import numpy as np
import pysindy as ps
from scipy.integrate import solve_ivp

# Define damped pendulum system
def damped_pendulum(t, state, g, L, gamma):
    theta, omega = state
    dtheta_dt = omega
    domega_dt = - (g / L) * np.sin(theta) - gamma * omega
    return [dtheta_dt, domega_dt]

# Generate data
g, L, gamma = 9.81, 1.0, 0.1  # Gravity, length, damping coefficient
t = np.linspace(0, 10, 100)   # Time array
theta0 = [np.pi / 4, 0]       # Initial conditions

sol = solve_ivp(damped_pendulum, (t[0], t[-1]), theta0, t_eval=t, args=(g, L, gamma))
X = sol.y.T  # theta, omega
dXdt = np.gradient(X, axis=0, edge_order=2) / (t[1] - t[0])  # Numerical derivatives

# Define candidate functions: Polynomial + Trigonometric (sin, cos)
library = ps.PolynomialLibrary(degree=2) + ps.FourierLibrary(n_frequencies=1)  

# Define constraint matrix (shape: number of constraints × number of features)
# Ensure that only sin(theta) is present and omega dependency remains
constraint = np.array([[0, 0, 0, 0, 1],  # Force sin(theta) presence
                       [0, 1, 0, 0, 0]]) # Ensure omega dependency

# Fit SINDy model with constraints
model = ps.SINDy(feature_library=library, constraint=constraint)
model.fit(X, t=t)

# Print identified equation
model.print()`
